lib/autoupdate/agent/setup.go

```diff
@@ -211,16 +216,34 @@ func (ns *Namespace) Setup(ctx context.Context) error {
 	if err != nil {
 		return trace.Wrap(err, "failed to write teleport-update systemd config files")
 	}
-	svc := &SystemdService{
+	timer := &SystemdService{
 		ServiceName: filepath.Base(ns.updaterTimerFile),
 		Log:         ns.log,
 	}
-	if err := svc.Sync(ctx); err != nil {
+	if err := timer.Sync(ctx); err != nil {
 		return trace.Wrap(err, "failed to sync systemd config")
 	}
-	if err := svc.Enable(ctx, true); err != nil {
+	if err := timer.Enable(ctx, true); err != nil {
 		return trace.Wrap(err, "failed to enable teleport-update systemd timer")
 	}
+	if ns.name == "" {
+		oldTimer := &SystemdService{
+			ServiceName: deprecatedTimerName,
+			Log:         ns.log,
+		}
+		// If the old teleport-upgrade script is detected, disable it to ensure they do not interfere.
+		// Note that the schedule is also set to nop by the Teleport agent -- this just prevents restarts.
+		enabled, err := oldTimer.IsEnabled(ctx)
+		if err != nil {
+			return trace.Wrap(err, "failed to determine if deprecated teleport-upgrade systemd timer is enabled")
```

Comment:
Should we log but keep going in this case?

If yes I suggest refactoring this branch into its own function, which errors normally, and log/swallow the error here.

Reply:
If we're unable to execute `IsEnabled`, something is seriously wrong and we should quit immediately. These functions only fail if:
- systemd is not available (`teleport-update` is unsupported), or
- `teleport-update` is unable to execute commands (and thus further operations will fail)